wal/wal.go

```diff
@@ -321,23 +323,20 @@ func (w *WAL) cut() error {
 	if err := w.sync(); err != nil {
 		return err
 	}
-	if err := w.f.Close(); err != nil {
-		return err
-	}
 
-	fpath := path.Join(w.dir, walName(w.seq+1, w.enti+1))
+	fpath := path.Join(w.dir, walName(w.seq()+1, w.enti+1))
 	ftpath := fpath + ".tmp"
 
 	// create a temp wal file with name sequence + 1, or truncate the existing one
-	ft, err := os.OpenFile(ftpath, os.O_WRONLY|os.O_APPEND|os.O_CREATE|os.O_TRUNC, 0600)
```

Comment:
The `O_TRUNC` seems necessary. If etcd is killed it could come back up, try to open the temp file, append to stale (possibly garbage) data, and potentially emit a corrupted WAL file.

According to Xiang the `O_APPEND`s were "for convenience" to avoid having to seek to the end of the file.